### PR TITLE
[#5159] core(feat): Add position support for column entity

### PR DESCRIPTION
**Pull request by jerryshao**
### What changes were proposed in this pull request?

This is a follow-up PR to support storing position for column entity.

### Why are the changes needed?

for future use.

Fix: #5159 

### Does this PR introduce _any_ user-facing change?

No.

### How was this patch tested?

UTs.

### CODE REVIEW

**Comment by jerryshao** [comment]
@yuqi1129 @mchades @jerqi please help to review this PR, thanks!

File: scripts/h2/schema-0.7.0-h2.sql [hunk 1]
```diff
@@ -93,6 +93,7 @@ CREATE TABLE IF NOT EXISTS `table_column_version_info` (
     `table_version` INT UNSIGNED NOT NULL COMMENT 'table version',
     `column_id` BIGINT(20) UNSIGNED NOT NULL COMMENT 'column id',
     `column_name` VARCHAR(128) NOT NULL COMMENT 'column name',
+    `column_position` INT UNSIGNED NOT NULL COMMENT 'column position',
```
Comment (yuqi1129) [inline scripts/h2/schema-0.7.0-h2.sql:96]: Is the value start from `0` or `1`? Please add this description here.

**Review by yuqi1129 — COMMENTED** [review]
Just a minor one, others look good to me.

File: scripts/h2/schema-0.7.0-h2.sql (hunk 1 above)
Comment (jerryshao) [reply scripts/h2/schema-0.7.0-h2.sql:96]: Starting from 0, I will add some comments.

**Review by yuqi1129 — APPROVED** [review]
LGTM

**Pull request merged**